test/system/distr_tests.cc

```diff
@@ -135,6 +135,24 @@ namespace detail {
 		});
 	}
 
+	TEST_CASE_METHOD(test_utils::runtime_fixture, "runtime-shutdown graph printing does not break when encountering a finished reduction",
```

Comment:
We won't notice if this test ever becomes obsolete when/if we change the way graph printing is handled (e.g. by not tying it directly to log level). You could assert that the graphs are being printed, see `log_capture` in device_selection_tests.cc.

Reply:
Good point! I changed that test to use `log_capture` and prepended a commit that guarantees stable log captures independent of the `CELERITY_LOG_LEVEL` setting (this broke the `device_selection_tests` before when the global log level was set below `info`).